Fixes #1651: Added StoryTextSizeFragment

MohamedMedhat1998 opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
- Fixes #1651
- Added **StoryTextSizeFragment** to the **StoryTextSizeActivity**
## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

MohamedMedhat1998 committed:
-Added "StoryTextSizeFragment"

rt4914 commented on app/src/main/java/org/oppia/app/options/OptionControlsViewModel.kt:
```diff
@@ -60,7 +60,7 @@ class OptionControlsViewModel @Inject constructor(
     itemViewModelList.clear()
 
     val optionsStoryTextViewViewModel =
```
Rename this variable to `optionsStoryTextSizeViewModel `

MohamedMedhat1998 replied:
Done

rt4914 commented on app/src/main/java/org/oppia/app/options/StoryTextSizeActivityPresenter.kt:
```diff
@@ -18,98 +14,21 @@ class StoryTextSizeActivityPresenter @Inject constructor(
   private var fontSize: String = getStoryTextSize(StoryTextSize.SMALL_TEXT_SIZE)
```
Default should be `MEDIUM_TEXT_SIZE `

rt4914 replied:
Also, check if you still need this? Because this is mainly used to preserve the configuration change which is needed for an activity but during configuration change fragment should handle this now and not the activity.

MohamedMedhat1998 replied:
removed `getStoryTextSize` function because it is no longer needed

rt4914 commented on app/src/main/java/org/oppia/app/options/StoryTextSizeActivityPresenter.kt:
```diff
@@ -18,98 +14,21 @@ class StoryTextSizeActivityPresenter @Inject constructor(
-        }
-      })
+    activity.setContentView(R.layout.story_text_size_activity)
+    val storyTextSizeFragment = StoryTextSizeFragment.newInstance(prefSummaryValue)
```
Please check if the fragment already exists or not, similar to `StoryActivityPresenter`

rt4914 replied:
This should be implemented in your other two PRs too.

MohamedMedhat1998 replied:
Done

rt4914 commented on app/src/main/java/org/oppia/app/options/StoryTextSizeFragment.kt:
```diff
@@ -0,0 +1,45 @@
+private const val KEY_STORY_TEXT_SIZE_PREFERENCE_SUMMARY_VALUE =
+  "STORY_TEXT_SIZE_PREFERENCE_SUMMARY_VALUE"
+
+/** The fragment to change the Text size of the Story content in the app. */
```
/** The fragment to change the text size of the story content in the app. */

MohamedMedhat1998 replied:
done

rt4914 commented on app/src/main/java/org/oppia/app/options/StoryTextSizeFragmentPresenter.kt:
```diff
@@ -0,0 +1,131 @@
+
+/** The presenter for [StoryTextSizeFragment]. */
+class StoryTextSizeFragmentPresenter @Inject constructor(private val fragment: Fragment) {
+  private var fontSize: String = getStoryTextSize(StoryTextSize.SMALL_TEXT_SIZE)
```
Change the default font to `MEDIUM_TEXT_SIZE`

MohamedMedhat1998 replied:
done

rt4914 commented on app/src/main/java/org/oppia/app/options/StoryTextSizeFragmentPresenter.kt:
```diff
@@ -0,0 +1,131 @@
+    fontSize = storyTextSize
+    updateTextSize(fontSize)
+
+    // The non-null receiver (?) is introduced here because the toolbar will be removed in the multipane
```
Please link this to a proper issue number as a TODO.

MohamedMedhat1998 replied:
Updated

rt4914 commented on app/src/main/java/org/oppia/app/options/StoryTextSizeFragmentPresenter.kt:
```diff
@@ -0,0 +1,131 @@
+    }
+  }
+
+  fun getStoryTextSizeInFloat(storyTextSize: StoryTextSize): Float {
```
Please add a TODO for this function improvement.
`// TODO(#1584): Update this function to use multiplier instead of direct font size.`

MohamedMedhat1998 replied:
Done

rt4914 reviewed (changes requested):
The implementation is nice but there will a lot of changes.

Also, as discussed in our last 1:1, create a separate text file `StoryTextSizeFragmentTest` and shift the test cases related to `StoryTextSize` correctly to this new test class and also make sure that both the files (`StoryTextSizeFragmentTest` and `OptionsFragmentTest`) passes all the tests.

rt4914 commented:
Also, please file a proper issue for these PRs, we should not have a PR without issue.

rt4914 commented:
> The implementation is nice but there will a lot of changes.
> 
> Also, as discussed in our last 1:1, create a separate text file `StoryTextSizeFragmentTest` and shift the test cases related to `StoryTextSize` correctly to this new test class and also make sure that both the files (`StoryTextSizeFragmentTest` and `OptionsFragmentTest`) passes all the tests.

This should be implemented in your other two PRs too.

MohamedMedhat1998 committed:
Merge remote-tracking branch 'origin/develop' into options-text-size-fragment

MohamedMedhat1998 commented:
> The implementation is nice but there will a lot of changes.
> 
> Also, as discussed in our last 1:1, create a separate text file `StoryTextSizeFragmentTest` and shift the test cases related to `StoryTextSize` correctly to this new test class and also make sure that both the files (`StoryTextSizeFragmentTest` and `OptionsFragmentTest`) passes all the tests.

Done

MohamedMedhat1998 commented:
> > The implementation is nice but there will a lot of changes.
> > Also, as discussed in our last 1:1, create a separate text file `StoryTextSizeFragmentTest` and shift the test cases related to `StoryTextSize` correctly to this new test class and also make sure that both the files (`StoryTextSizeFragmentTest` and `OptionsFragmentTest`) passes all the tests.
> 
> This should be implemented in your other two PRs too.

I'll do so once the approach in this PR is approved

rt4914 commented on app/src/main/java/org/oppia/app/options/StoryTextSizeFragment.kt:
```diff
@@ -0,0 +1,45 @@
+  lateinit var storyTextSizeFragmentPresenter: StoryTextSizeFragmentPresenter
+
+  companion object {
+    internal const val STORY_TEXT_SIZE_FRAGMENT_TAG = "STORY_TEXT_SIZE_FRAGMENT_TAG"
```
This is not getting used anywhere. Remove this if not needed.

MohamedMedhat1998 replied:
Done

rt4914 commented on app/src/main/java/org/oppia/app/options/StoryTextSizeFragmentPresenter.kt:
```diff
@@ -0,0 +1,132 @@
+    fontSize = storyTextSize
+    updateTextSize(fontSize)
+
+    // TODO(#1200): stop the toolbar functionality in the multipane (add non-null receiver (?))
```
nit:`// TODO(#1200): Stop the toolbar functionality in the multipane (add non-null receiver (?)).`

MohamedMedhat1998 replied:
Done

rt4914 commented on app/src/test/java/org/oppia/app/testing/OptionsFragmentTest.kt:
```diff
@@ -0,0 +1,72 @@
+import org.oppia.app.options.StoryTextSizeActivity
+import org.oppia.app.recyclerview.RecyclerViewMatcher.Companion.atPositionOnView
+
+private const val KEY_STORY_TEXT_SIZE_PREFERENCE_TITLE = "STORY_TEXT_SIZE_PREFERENCE"
```
Remove this from here, instead define this in activitiy's companion object and use it it here using, <AcgtivityName>.<CONSTANT_NAME>

MohamedMedhat1998 replied:
Done

rt4914 commented on app/src/test/java/org/oppia/app/testing/StoryTextSizeFragmentTest.kt:
```diff
@@ -0,0 +1,80 @@
+
+  @Test
+  fun changeTextSizeToLarge_changeConfiguration_checkTextSizeLargeIsSelected() {
+    launch<StoryTextSizeActivity>(createStoryTextSizeActivityIntent("small")).use {
```
nit:
`Small`

MohamedMedhat1998 replied:
Done

rt4914 commented on app/src/test/java/org/oppia/app/testing/StoryTextSizeFragmentTest.kt:
```diff
@@ -0,0 +1,80 @@
+import org.oppia.app.options.StoryTextSizeActivity
+import org.oppia.app.utility.OrientationChangeAction.Companion.orientationLandscape
+
+@RunWith(AndroidJUnit4::class)
```
Shift this file to a separate `options` package instead of `testing`

MohamedMedhat1998 replied:
Done, moved to `testing/options`

rt4914 commented on app/src/test/java/org/oppia/app/testing/OptionsFragmentTest.kt:
```diff
@@ -0,0 +1,72 @@
+private const val KEY_STORY_TEXT_SIZE_PREFERENCE_TITLE = "STORY_TEXT_SIZE_PREFERENCE"
+
+@RunWith(AndroidJUnit4::class)
+class OptionsFragmentTest {
```
Shift this file to a separate `options` package instead of `testing`

MohamedMedhat1998 replied:
Done, moved to `testing/options`

rt4914 reviewed (changes requested):
PTAL

MohamedMedhat1998 committed:
-Added new test cases for "StoryTextSizeFragment"
-Updated the current test cases in "OptionsFragmentTest" related to "StoryTextSizeFragment"

MohamedMedhat1998 committed:
-Some nit changes

MohamedMedhat1998 committed:
Merge remote-tracking branch 'origin/develop' into options-text-size-fragment

rt4914 reviewed (approved):
LGTM
There is one test cases which is failing from previous implementation but I know it will get fixed in your next PRs.
`testOptionFragment_changeConfiguration_checkAudioLanguageIsHindi()`

MohamedMedhat1998 committed:
-Some nit changes
